src/pkg/util/fs/mount/system.go

```diff
@@ -0,0 +1,84 @@
+// Copyright (c) 2018, Sylabs Inc. All rights reserved.
+// This software is licensed under a 3-clause BSD license. Please consult the
+// LICENSE file distributed with the sources of this project regarding your
+// rights to use or distribute this software.
+
+package mount
+
+import (
+	"fmt"
+)
+
+// hookFn describes function prototype for function
+// to be called before/after mounting a tag list
+type hookFn func(*System) error
+
+// mountFn describes function prototype for function responsible
+// of mount operation
+type mountFn func(*Point) error
```

Comment:
I think this should be exported (`MountFn`). Reasoning is so that when someone is developing using this library, they can see exactly what the type they are supposed to use is (via GoDoc or VSCode, etc...)